aiohttp/web_utils.py

```diff
@@ -0,0 +1,57 @@
+import asyncio
+
+from aiohttp.web_exceptions import HTTPMovedPermanently, HTTPNotFound
+from aiohttp.web_urldispatcher import SystemRoute
+
+__all__ = ('normalize_path_middleware',)
+
+
+def normalize_path_middleware(*, merge_slashes=True, append_slash=True,
+                              redirect_class=HTTPMovedPermanently):
+
+    @asyncio.coroutine
+    def factory(app, handler):
+
+        @asyncio.coroutine
+        def middleware(request):
+            try:
+                return (yield from handler(request))
+            except HTTPNotFound as exc:
```

Comment:
The code seems to work as is. I have a test case here: https://github.com/sloria/aiohttp_utils/blob/79b38453f924f9a569b6a06f5f3a584d3ecf11da/tests/test_path_norm.py#L116-L121 (setup here: https://github.com/sloria/aiohttp_utils/blob/79b38453f924f9a569b6a06f5f3a584d3ecf11da/tests/test_path_norm.py#L22-L30)